include/maliput/common/maliput_throw.h

```diff
@@ -162,15 +173,33 @@ void Throw(const char* condition, const char* func, const char* file, int line)
         error_message.c_str(), __func__, __FILE__, __LINE__);                              \
   } while (0)
 
-/// @def MALIPUT_VALIDATE
-/// Used to validate that a @p pred-icate passed into a function or method is
-/// true; if not, an exception with @p message is thrown.
-#define MALIPUT_VALIDATE(pred, message)                                                                            \
-  do {                                                                                                             \
-    if (!(pred)) {                                                                                                 \
-      ::maliput::common::internal::Throw<maliput::common::assertion_error>(std::string(message).c_str(), __func__, \
-                                                                           __FILE__, __LINE__);                    \
-    }                                                                                                              \
+/// Evaluates @p condition and iff the value is false will throw an
+/// @p err_type with a message showing at least the
+/// condition text, function name, file, and line.
+#define MALIPUT_THROW_RULES_UNLESS(condition, err_type)                                       \
+  do {                                                                                        \
+    if (!(condition)) {                                                                       \
+      ::maliput::common::internal::Throw<err_type>(#condition, __func__, __FILE__, __LINE__); \
+    }                                                                                         \
+  } while (0)
```

Comment:
What's the difference with MALIPUT_RULES_VALIDATE?
Let's stick to one or the other.

Reply:
Added a TODO to deprecate `MALIPUT_THROW_UNLESS` in favor of `MALIPUT_VALIDATE`.
Currently, `MALIPUT_VALIDATE` supports custom messages, which `MALIPUT_THROW_UNLESS` doesn't, so it will need some work.